alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDocLevelMonitorFanOutAction.kt

```diff
@@ -207,6 +207,22 @@ class TransportDocLevelMonitorFanOutAction
         listener: ActionListener<DocLevelMonitorFanOutResponse>
     ) {
         try {
+            // handle situations where coordinator node is not on the same opensearch version or
+            // from the same domain instance and return no-op response instead of failing entire fan-out
+            if (request.indexExecutionContext == null) {
```

Comment:
Checking this specific field for null seems brittle. Is there a more definitive way to see if the deserialization failed?

Reply:
+1